### PR TITLE
vim: Add delete action to HelixNormal mode

### PR DESCRIPTION
**Pull request by leroycep**
Related issue: https://github.com/zed-industries/zed/issues/4642

Release-Notes:

* N/A

### CODE REVIEW

**Comment by leroycep** [comment]
@cla-bot check

File: crates/vim/src/helix.rs [hunk 1]
```diff
@@ -226,6 +227,33 @@ impl Vim {
...
+        self.store_visual_marks(cx);
+        self.update_editor(cx, |vim, editor, cx| {
+            // Fixup selections so they have helix's semantics.
+            // Specifically:
+            //  - Make sure that each cursor acts as a 1 character wide selection
+            editor.transact(cx, |editor, cx| {
+                editor.change_selections(Some(Autoscroll::fit()), cx, |s| {
+                    s.move_with(|map, selection| {
+                        if selection.is_empty() && !selection.reversed {
+                            *selection.end.column_mut() += 1;
```
Comment (ConradIrwin) [inline crates/vim/src/helix.rs:241]: @leroycep it is not safe to +1 to a column without clipping it, as you may put the cursor in the middle of a multlibyte utf8 sequence.

It'd be better to do `selection.end = movement::right(selection.end)`

Not sure how soon we want to support `.` for helix,

Comment (leroycep) [reply crates/vim/src/helix.rs]: updated to use `movement::right`

Comment (ConradIrwin) [reply crates/vim/src/helix.rs]: I think that means we don't need the second condition as right at the end of the line will move the start of the next?

Comment (leroycep) [reply crates/vim/src/helix.rs]: Yep, that seems to be the case. The tests I added still pass with it removed.

Does the `!selection.reversed` check need to be there? I'm not sure if it's possible for a selection to be reversed and empty.

Comment (ConradIrwin) [reply crates/vim/src/helix.rs]: It's not needed, reversed is only if head < tail.

**Pull request merged**